lib/stones-spec/with_gbb_html_rendering.rb

```diff
@@ -1,15 +1,21 @@
 module StonesSpec
   module WithGbbHtmlRendering
-    def get_html_board(gbb_representation)
+    def get_html_board(caption, gbb_representation)
       identity = write_tempfile 'program {}', '.gbs'
       board = write_tempfile gbb_representation, '.gbb'
       board_html = Tempfile.new %w(gobstones.board .html)
 
       %x"#{Language::Gobstones.run(identity, board, board_html)}"
 
-      board_html.read
+      add_caption caption, board_html.read
     ensure
       [identity, board, board_html].compact.each(&:unlink)
     end
+
+    private
+
+    def add_caption(caption, board_html)
+      board_html.sub '<table class="gbs_board">', "<table class=\"gbs_board\">\n<caption>#{caption}</caption>"
```

Comment:
In order to be consistent with previous changes, I would name it `with_caption` instead of `add_caption`